src/modules/Game.class.js

```diff
@@ -50,19 +64,189 @@ class Game {
    * `win` - the game is won;
    * `lose` - the game is lost
    */
-  getStatus() {}
+  getStatus() {
+    switch (true) {
+      case this.isGameWon:
+        this.status = 'win';
+        break;
+      case this.isGameLose:
+        this.status = 'lose';
+        break;
+      case this.isGameActive:
+        this.status = 'playing';
+        break;
+      default:
+        this.status = 'idle';
+        break;
+    }
 
-  /**
-   * Starts the game.
-   */
-  start() {}
+    return this.status;
+  }
 
-  /**
-   * Resets the game.
-   */
-  restart() {}
+  start() {
+    this.board = JSON.parse(JSON.stringify(this.initialBoard));
+
+    this.isGameActive = true;
+    this.placeNewCell();
+    this.placeNewCell();
+  }
+
+  restart() {
+    this.board = JSON.parse(JSON.stringify(this.initialBoard));
+    this.boardScore = 0;
+
+    this.currentStatus = 'idle';
+    this.isGameActive = false;
+    this.isGameWon = false;
+    this.isGameLose = false;
+  }
+
+  placeNewCell() {
+    let rRow, rColumn;
+
+    do {
+      rRow = Math.floor(Math.random() * 4);
+      rColumn = Math.floor(Math.random() * 4);
+    } while (this.board[rRow][rColumn] !== 0);
+
+    this.board[rRow][rColumn] = this.createNewCell();
+  }
+
+  createNewCell() {
+    const randomValue = Math.random();
+
+    return randomValue < 0.1 ? 4 : 2;
+  }
+
+  move(direction) {
+    if (!this.isGameActive) {
+      return;
+    }
+
+    const numCols = this.board[0].length;
+    let currentTable = JSON.parse(JSON.stringify(this.board));
+    let addScore = 0;
+
+    const transpose = (table) => {
+      return table[0].map((_, colIndex) => table.map((row) => row[colIndex]));
+    };
+
+    const reverseRow = (table) => {
+      return table.map((row) => row.slice().reverse());
+    };
 
-  // Add your own methods here
+    const moveTable = (table) => {
+      const newTable = table.map((row) => {
+        let newRow = row.filter((num) => num !== 0);
+        const zeroToAdd = numCols - newRow.length;
+
+        newRow = [...Array(zeroToAdd).fill(0), ...newRow];
+
+        for (let i = newRow.length; i >= 0; i--) {
+          if (newRow[i - 1] === newRow[i]) {
+            newRow[i - 1] *= 2;
+            newRow[i] = 0;
+            addScore += newRow[i - 1];
+            i--;
+          }
+        }
+
+        newRow = newRow.filter((num) => num !== 0);
+
+        const zerosToAddEnd = numCols - newRow.length;
+
+        newRow = [...Array(zerosToAddEnd).fill(0), ...newRow];
+
+        return newRow;
+      });
+
+      return newTable;
+    };
+
+    const isGameOver = () => {
```

Comment:
Do not define this function in `move` method. Define it as a separate method. Check other similar cases